### ISSUE TITLE
Resampling seems to shift the image

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#612]
see below (downsampling by factor 4 along z):
![output_h3a4nq](https://cloud.githubusercontent.com/assets/2482071/10646355/a58e5f94-7800-11e5-9c85-b8c0e2fc405c.gif)

**Comment by SaraDupont** [comment]
I reproduced it with a non straighten segmentation of a T2, but I don't know where it comes from: the slices values seems to match (ie the index of the first and last segmentation slice match between the original image and the resampled image taking the resampling factor into account: in the segmentation I down sampled with a factor 4, the last slice in the last segmentation slice (in axial plane) is slice # 221  and in the resampled image it's slice # 55). 

The shift is more obvious with down sampling than with up sampling so maybe it's due to the approximation of the interpolation (if so, I don't se what can we  do about it).

**Comment by SaraDupont** [comment]
I think either the shift comes from dipy (we use ```dipy.align.aniso2iso.reslice()``` to resample) or there is actually no shift (or a small one due to interpolation that we can't avoid since interpolation can't be perfect) and the preview gives a wrong impression (I don't know why it would though)

**Comment by jcohenadad** [comment]
i think there is definitely a shift. Another example below: 0.3x0.3x2.75mm --> 1x1x1mm

![output_qsq7gx](https://cloud.githubusercontent.com/assets/2482071/14294056/3d0da98c-fb3d-11e5-9fcf-0ef559db9f7a.gif)

**Comment by jcohenadad** [comment]
fixed by switching to ```nipy.algorithms.resample``` method, which operates in the world coordinate, and adjusting for half-voxel shift of the image by applying translation to the transformation matrix (input of resample function).

**Comment by jcohenadad** [comment]
Issue is back! See below with commit f7ebd4620218c7c0763312d58474c7de5a06c49c
![anim](https://user-images.githubusercontent.com/2482071/37195726-608dd39c-23b7-11e8-949f-098c753b494a.gif)

**Comment by joshuacwnewton** [comment]
I'm going to close this issue for two main reasons:

1. This issue is more than 5 years old, and the data that shows the shift ([#612 (comment](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/612#issuecomment-371735783)) is likely inaccessible/hard to access at this point. In lieu of a specific image, I tried 4x downsampling (trying vox/mm/factor) with some sample images (both isotropic and anisotropic), but I don't see any shifts. So, unless we're able to access the original image, this behavior will be difficult to reproduce.
2. More importantly, however, I did a quick investigation into the resampling code, comparing 2018 to 2023:
   - SCT's outer wrappers (`sct_resample`, `resample_file()`) haven't changed in a meaningful way.
   - However, within the inner wrapper (`resample_image()`), I did notice a suspicious line that was present in 2018, but no longer present in 2023:
   https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/f7ebd4620218c7c0763312d58474c7de5a06c49c/spinalcordtoolbox/resample/nipy_resample.py#L86-L87 
   This line was removed during a major refactor in 2019 ([#2157](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2157), https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2157/commits/e95b18df71e97346268825ac2cb627d33fd9ff85). I have a good feeling about this change; our code now has less "manual" logic, and so I feel that it's less likely that SCT's code is causing unwanted shifts.
   - Apart from that, the only other place the bug could be from is the third party resampling code. This has changed _significantly_ over the last 5 years (`nipy.resample()` -> `nib.resample_from_to()`). So, if this was an upstream bug, there is a good chance the old behavior no longer applies to the new resampling code.

tl;dr, I think it's somewhat likely that this bug was addressed by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2157. And, because of the difficulty in reproducing, I think the practical thing to do is close this issue for now, and simply re-open if this behavior is encountered again.
